app/billing/rest.py

```diff
@@ -106,17 +106,34 @@ def get_free_sms_fragment_limit(service_id):
         financial_year_start = request.args.get('financial_year_start')
 
     if financial_year_start is None:
-        results = dao_get_all_free_sms_fragment_limit(service_id)
-
-        if len(results) == 0:
-            raise InvalidRequest('no annual billing information for this service', status_code=404)
-        return jsonify(data=[row.serialize_free_sms_items() for row in results]), 200
+        # return a list of all entries related to the service
+        sms_list = dao_get_all_free_sms_fragment_limit(service_id)
+        if len(sms_list) == 0:
+            raise InvalidRequest('no free-sms-fragment-limit list for this service in DB', status_code=404)
```

Comment:
Rather than erroring with `this service in DB`, you could mention the service affected, `service xxx` as not sure how `in DB` would help